dynamic_programming/subset_generation

```diff
@@ -0,0 +1,62 @@
+# python program to print all subset combination of n element in given set of r element .
+
+#arr[]  ---> Input Array
+#data[] ---> Temporary array to store current combination
+# start & end ---> Staring and Ending indexes in arr[]
+# index  ---> Current index in data[]
+#r ---> Size of a combination to be printed 
+
+
+def combinationUtil(arr,n,r,index,data,i):
+#Current combination is ready to be printed,
+# print it
+ if(index== r):
+  for j in range(r):
+   print(data[j],end=" ")
+  print(" ")
+  return
+
+#  When no more elements are there to put in data[]
+
```

Comment:
Remove useless empty lines. You can write this:
```python

# comment
def foo():
   pass

# comment
def boo():
  pass
```

Above each comment you can write a empty line. For example:
```python
def combinationUtil(arr,n,r,index,data,i):
      
#Current combination is ready to be printed,
# print it
 if(index== r):
  for j in range(r):
   print(data[j],end=" ")
  print(" ")
  return

#  When no more elements are there to put in data[]
 if(i>=n):
  return

#current is included, put next at next
# location 
 data[index]=arr[i]
 combinationUtil(arr,n,r,index+1,data,i+1)
```